compiler/AST/expr.cpp

```diff
@@ -5808,12 +5816,31 @@ GenRet CallExpr::codegenPrimitive() {
 GenRet CallExpr::codegenPrimMove() {
   GenRet ret;
 
+  GenRet specRet;
   if (get(1)->typeInfo() == dtVoid) {
     ret = get(2)->codegen();
 
   // Is the RHS a primop with special case handling?
-  } else if (codegenPrimMoveRhsIsSpecialPrimop() == true) {
-    // Codegen has been handled.  No more activity here.
+  } else if (codegenIsSpecialPrimitive(get(1), get(2), specRet)) {
+    CallExpr* rhsCe = toCallExpr(get(2));
+    if(rhsCe->isPrimitive(PRIM_GET_REAL) || rhsCe->isPrimitive(PRIM_GET_IMAG)) {
+
+      if (gGenInfo->cfile) {
+        std::string stmt = codegenValue(get(1)).c + " = ";
+
+        stmt += specRet.c;
+        stmt += ";\n";
+
+        gGenInfo->cStatements.push_back(stmt);
+      } else {
+#ifdef HAVE_LLVM
+        codegenStoreLLVM(*specRet, get(1));
+#endif
+      }
```

Comment:
If you inherited it from the previous function - don't feel obligated to fix it now - but of course I would be happy if you did fix it. Anyway it would help at least to put a comment indicating that this code doesn't necessarily need to stay here.